### DISCUSSION TITLE
All-in-One build

### DISCUSSION THREAD

**Discussion by nricciardi** [KaTeX/KaTeX#3924] [Q&A]
I would insert in one single html page KaTeX functionality. I have already tried using script and link tag to include `katex.min.js` and `katex.min.css`, but without fonts the math in page doesn't render correctly.
How can i fix?

**Comment by edemaine** [comment]
You need the font files for KaTeX to render correctly. Does this answer your question?

**Reply by nricciardi** [reply]
Unfortunately, no.
I would have a single local html page where is possible to render "mathematical symbols". I have tried to download built version of KaTeX, included `katex.min.js` e `katex.min.css` as `head` `link` and  `script`, but without fonts KaTeX doesn't work correctly. My question is **is there a way to insert KaTeX fonts in html just like `min.js` and `min.css`?** So that I can use html page "offline".

**Reply by edemaine** [reply]
I think you could inline the font files in base64 into the CSS, as hinted at e.g. [here](https://stackoverflow.com/questions/26867893/converting-and-rendering-web-fonts-to-base64-keep-original-look).

**Reply by nricciardi** [reply]
I have also tried, but I think that fonts are injected "at run time" from `fonts` directory, so it is pointless insert them in base 64...

**Reply by edemaine** [reply]
`katex.min.css` has content like `@font-face{font-family:KaTeX_AMS;font-style:normal;font-weight:400;src:url(fonts/KaTeX_AMS-Regular.woff2) format("woff2")}`. Instead of `url(fonts/KaTeX_AMS-Regular.woff2)` you should be able to replace that with a base-64 encoded version of that file. (You can probably get away with just one extension version of each font.)

**Reply by nricciardi** [reply]
Thanks, I will try!
Maybe in future release we could add this special `css`
